.github/workflows/experimental.yml

```diff
@@ -63,12 +63,16 @@ jobs:
           key: ccache-benchmark-${{ github.sha }}
           restore-keys: |
             ccache-benchmark-
+
       - name: "Checkout Repo"
+        if: ${{ github.event_name == 'pull_request' }}
```

Comment:
@majetideepak @assignUser Folks, would you help review .yml changes?

Reply:
@majetideepak @assignUser Do you have any comment for the change in this PR? Thanks!

Reply:
@rui-mo can you clarify the changes to the compile step?
Are we now building artifacts for a pull request?

Reply:
I found the fuzzer test ran on main branch instead of on the pull request, which causes the change in this PR could not be verified in the workflow. I wonder if it is intended to run the fuzzer test for each pull request. Could you give me some inputs here? Thanks.

Reply:
The current flow seems to be that the compile job builds and uploads the artifacts to the main branch and the other jobs run the fuzzer artifacts from the main branch.
There will be a race condition if we upload artifacts per PR as they likely will overwrite each other.
@kgpai can you provide context on this?